solidity/contracts/system/KeepFactorySelection.sol

```diff
@@ -150,29 +170,80 @@ library KeepFactorySelection {
         );
     }
 
+    /// @notice Returns KEEP staked factory address. If factories lock is not set
+    /// it calls the KEEP staked vendor to obtain the latest versions of the factory.
+    function getKeepStakedFactory(Storage storage _self)
+        internal
+        view
+        returns (IBondedECDSAKeepFactory)
+    {
+        if (_self.factoriesVersionsLock) {
+            return _self.keepStakeFactory;
+        } else {
+            IBondedECDSAKeepFactory keepStakeFactory = IBondedECDSAKeepFactory(
+                _self.keepStakeVendor.selectFactory()
+            );
+
+            require(
+                address(keepStakeFactory) != address(0),
+                "Vendor returned invalid factory address"
+            );
+
+            return keepStakeFactory;
+        }
+    }
+
+    /// @notice Returns ETH-stake based factory address. If factories lock is not set
+    /// it calls the ETH staked vendor to obtain the latest versions of the factory.
+    function getFullyBackedFactory(Storage storage _self)
+        internal
+        view
+        returns (IBondedECDSAKeepFactory)
+    {
+        if (_self.factoriesVersionsLock) {
+            return _self.ethStakeFactory;
+        } else {
+            IBondedECDSAKeepFactory ethStakeFactory = IBondedECDSAKeepFactory(
+                _self.ethStakeVendor.selectFactory()
+            );
+
+            require(
+                address(ethStakeFactory) != address(0),
+                "Vendor returned invalid factory address"
+            );
+
+            return ethStakeFactory;
+        }
+    }
+
     /// @notice Sets the address of the fully backed, ETH-stake based keep
     /// factory. KeepFactorySelection can work without the fully-backed keep
     /// factory set, always selecting the default KEEP-stake-based factory.
     /// Once both fully-backed keep factory and factory selection strategy are
     /// set, KEEP-stake-based factory is no longer the default choice and it is
     /// up to the selection strategy to decide which factory should be chosen.
     /// @dev Can be called only one time!
-    /// @param _fullyBackedFactory Address of the fully-backed, ETH-stake based
-    /// keep factory.
-    function setFullyBackedKeepFactory(
+    /// @param _fullyBackedVendor Address of the fully-backed, ETH-stake based
+    /// keep vendor.
+    function setFullyBackedKeepVendor(
```

Comment:
`setFullyBackedKeepFactoryVendor` ?

Reply:
I'm not sure, it's `BondedECDSAKeepVendor` not `BondedECDSAKeepFactoryVendor`.

Reply:
Yep, you are right. Resolving.